lib/vmdb/settings.rb

```diff
@@ -88,13 +88,28 @@ def self.build_template
     private_class_method :build_template
 
     def self.build(resource)
-      build_template.tap do |settings|
-        settings.add_source!(DatabaseSource.new(resource))
+      objects_in_settings_chain = resource ? resource.objects_in_settings_chain : []
+      build_without_local(objects_in_settings_chain).tap do |settings|
+        objects_in_settings_chain.each { |r| settings.add_source!(DatabaseSource.new(r)) }
         local_sources.each { |s| settings.add_source!(s) } if resource.try(:is_local?)
       end
     end
     private_class_method :build
 
+    def self.parent_settings(resource)
+      objects_in_settings_chain = resource.objects_in_settings_chain
+      objects_in_settings_chain.pop
```

Comment:
Or rather...after reading the next line you probably want `objects_in_settings_chain[0...-1]`